src/osd/PrimaryLogPG.cc

```diff
@@ -8602,6 +8552,67 @@ void PrimaryLogPG::finish_ctx(OpContext *ctx, int log_op_type, int result)
 	   << dendl;
   utime_t now = ceph_clock_now();
 
+  // Drop the reference if deduped chunk is modified
+  if (ctx->new_obs.oi.is_dirty() &&
+    (ctx->obs->oi.has_manifest() && ctx->obs->oi.manifest.is_chunked()) &&
+    // If a clone is creating, ignore dropping the reference for manifest object
+    !ctx->delta_stats.num_object_clones &&
+    ctx->new_obs.oi.size != 0 && // missing, redirect and delete
+    log_op_type != pg_log_entry_t::PROMOTE) {
+    object_manifest_t updated_chunks;
+    object_info_t &oi = ctx->obc->obs.oi;
+    updated_chunks.type = object_manifest_t::TYPE_CHUNKED;
+    // try to find out intersections
+    for (auto &p : oi.manifest.chunk_map) {
+      if (is_dedup_chunk(oi, p.second)) {
+	if (!ctx->clean_regions.is_clean_region(p.first, p.second.length) &&
+	    p.second.oid != hobject_t()) {
+	  chunk_info_t updated_chunk;
+	  updated_chunk.offset = p.second.offset;
+	  updated_chunk.length = p.second.length;
+	  updated_chunk.oid = p.second.oid;
+	  updated_chunks.chunk_map[p.first] = updated_chunk;
+	  updated_chunks.chunk_map[p.first].set_flag(chunk_info_t::FLAG_HAS_FINGERPRINT);
+	}
+      }
+    }
+
+    dout(20) << " intersected manifest " << updated_chunks.chunk_map << dendl;
+
+    if (updated_chunks.chunk_map.size()) {
```

Comment:
Find a way to factor this into a helper.

Reply:
The get_object_context will mean you can't unit test it, but finish_ctx is already huge, so I'd rather have dec_refcount_by_dirty or something.